File: Serene/Serene.Script/Imports/ServiceContracts/Administration.TranslationService.cs
namespace Serene.Administration
{
    using jQueryApi;
    using Serenity;
    using System;
    using System.Collections;
    using System.Collections.Generic;
    using System.Runtime.CompilerServices;

    [Imported, PreserveMemberCase]
    public partial class TranslationService
    {
        [InlineConstant] public const string BaseUrl = "Administration/Translation";
    
        [InlineCode("Q.serviceRequest('Administration/Translation/List', {request}, {onSuccess}, {options})")]
        public static jQueryXmlHttpRequest List(TranslationListRequest request, Action<ListResponse<TranslationItem>> onSuccess, ServiceCallOptions options = null)
        {
            return null;
        }
    
        [InlineCode("Q.serviceRequest('Administration/Translation/Update', {request}, {onSuccess}, {options})")]
        public static jQueryXmlHttpRequest Update(TranslationUpdateRequest request, Action<SaveResponse> onSuccess, ServiceCallOptions options = null)
        {
            return null;
        }
    
        [Imported, PreserveMemberCase]
        public static class Methods
        {
            [InlineConstant] public const string List = "Administration/Translation/List";
            [InlineConstant] public const string Update = "Administration/Translation/Update";
        }
    }
    
}

File: Serene/Serene.Web/Modules/Common/Imports/ServerTypings/Administration.TranslationService.ts
namespace Serene.Administration {
    export namespace TranslationService {
        export const baseUrl = 'Administration/Translation';

        export declare function List(request: TranslationListRequest, onSuccess?: (response: Serenity.ListResponse<TranslationItem>) => void, opt?: Serenity.ServiceOptions<any>): JQueryXHR;
        export declare function Update(request: TranslationUpdateRequest, onSuccess?: (response: Serenity.SaveResponse) => void, opt?: Serenity.ServiceOptions<any>): JQueryXHR;

        export namespace Methods {
            export declare const List: string;
            export declare const Update: string;
        }

        ['List', 'Update'].forEach(x => {
            (<any>TranslationService)[x] = function (r, s, o) { return Q.serviceRequest(baseUrl + '/' + x, r, s, o); };
            (<any>Methods)[x] = baseUrl + '/' + x;
        });
    }
}

